scp.py

```diff
@@ -355,9 +355,9 @@ def _recv_confirm(self):
         except SocketTimeout:
             raise SCPException('Timeout waiting for scp response')
         # slice off the first byte, so this compare will work in py2 and py3
-        if msg and msg[0:1] == b'\x00':
+        if msg and type(msg) == bytes and (repr(str(msg).startswith('\x00')) or repr(str(msg).endswith('\x00'))):
             return
-        elif msg and msg[0:1] == b'\x01':
+        elif msg and type(msg) == bytes and repr(str(msg).startswith('\x01')):
```

Comment:
Why check the type? Does paramiko's `recv()` ever return something else?

Reply:
Ya I don't think it does, but I was trying to be accommodating because I converted the bytes type to a string.

Reply:
You can check for a null byte without converting

Reply:
More than happy to accept a better implementation.  I was having trouble checking null byte at the end.

Reply:
I don't understand why you are calling `repr()` either?